src/backend/distributed/commands/truncate.c

```diff
@@ -48,6 +48,18 @@ static void AcquireDistributedLockOnRelations(List *relationIdList, LOCKMODE loc
 void
 ProcessTruncateStatement(TruncateStmt *truncateStatement)
 {
+	ListCell *relationCell = NULL;
+
+	/*
+	 * Lock relations to avoid them being dropped or their structure being
```

Comment:
Hmm, isn't `AccessShareLock ` is too weak for TRUNCATE? And, also the issue with this approach that we do not capture relations that are truncated via `TRUNCATE .. CASCADE` and partitioned relations. I guess that's one of main reasons that we use truncate triggers so that it cascades to related relations.

So, my question is that, do we really need to have these locks? I cannot see anything related to this in https://github.com/citusdata/citus/issues/2888 as well

Reply:
Removed this lock. Changed all `heap_open()` calls to `RangeVarGetRelid(..., NoLock, ...)` which doesn't have the offending assertion. This keeps the behaviour as before for this release without failing assertions in pg12. We will look into this closer in next release.